Example/AblyChatExample/Mocks/Mocks.swift

```diff
@@ -0,0 +1,177 @@
+        self.clientOptions = clientOptions
+    }
+}
+
+actor MockRoom: Room {
+    internal nonisolated let roomID: String
+    internal nonisolated let options: RoomOptions
+    
+    internal init(roomID: String, options: RoomOptions) {
+        self.roomID = roomID
+        self.options = options
+    }
+    
+    nonisolated var messages: any Messages {
+        MockMessages(clientID: "AblyTest", roomID: roomID)
+    }
+
+    nonisolated var presence: any Presence {
+        fatalError("Not yet implemented")
+    }
+
+    nonisolated var reactions: any RoomReactions {
+        MockRoomReactions(clientID: "AblyTest", roomID: roomID)
+    }
+
+    nonisolated var typing: any Typing {
+        MockTyping(clientID: "AblyTest", roomID: roomID)
+    }
+
+    nonisolated var occupancy: any Occupancy {
+        fatalError("Not yet implemented")
+    }
+
+    nonisolated var status: any RoomStatus {
+        fatalError("Not yet implemented")
+    }
+
+    func attach() async throws {
+        fatalError("Not yet implemented")
+    }
+
+    func detach() async throws {
+        fatalError("Not yet implemented")
+    }
+}
+
+actor MockMessages: Messages {
+    let clientID: String
+    let roomID: String
+    let channel: RealtimeChannel
+    
+    init(clientID: String, roomID: String) {
+        self.clientID = clientID
+        self.roomID = roomID
+        self.channel = MockRealtimeChannel()
+    }
+    
+    func subscribe(bufferingPolicy: BufferingPolicy) async -> MessageSubscription {
+        MessageSubscription(mockAsyncSequence: MockMessageSubscription(clientID: clientID, roomID: roomID)) {_ in 
+            MockMessagesPaginatedResult(clientID: self.clientID, roomID: self.roomID)
+        }
+    }
+    
+    func get(options: QueryOptions) async throws -> any PaginatedResult<Message> {
+        MockMessagesPaginatedResult(clientID: self.clientID, roomID: self.roomID)
+    }
+    
+    func send(params: SendMessageParams) async throws -> Message {
+        Message(timeserial: "\(Date().timeIntervalSince1970)",
+                clientID: self.clientID,
+                roomID: self.roomID,
+                text: params.text,
+                createdAt: Date(),
+                metadata: [:],
+                headers: [:])
+    }
+    
+    func subscribeToDiscontinuities() -> Subscription<ARTErrorInfo> {
+        fatalError("Not yet implemented")
+    }
+}
+
+actor MockRoomReactions: RoomReactions {
+    let clientID: String
+    let roomID: String
+    let channel: RealtimeChannel
+    
+    init(clientID: String, roomID: String) {
+        self.clientID = clientID
+        self.roomID = roomID
+        self.channel = MockRealtimeChannel()
+    }
+    
+    func send(params: RoomReactionParams) async throws {
+        _ = Reaction(type: "like",
+                     metadata: [:],
+                     headers: [:],
+                     createdAt: Date(),
+                     clientID: clientID,
+                     isSelf: true)
```

Comment:
Shouldn't these look like they're coming from other users? ditto for messages

Reply:
https://github.com/ably-labs/ably-chat-swift/pull/34/commits/7a7035984dc4bdcfebb276aac0bb18a7293e8196